lib/child_process.js

```diff
@@ -323,6 +324,11 @@ function normalizeSpawnArguments(file /*, args, options*/) {
   // Make a shallow copy so we don't clobber the user's options object.
   options = Object.assign({}, options);
 
+  if (options.shell === 'auto') {
```

Comment:
This logic could be moved down to https://github.com/timdp/node/blob/0bd9febb52756a8e2069d279703601bc017d5ab0/lib/child_process.js#L335, where the presence of the `shell` options and the windows check are already done for you.